esp/smc/SMCLib/TpWrapper.cpp

```diff
@@ -1674,6 +1674,45 @@ void CTpWrapper::getDropZoneMachineList(double clientVersion, bool ECLWatchVisib
 
 void CTpWrapper::getTpDropZones(double clientVersion, const char* name, bool ECLWatchVisibleOnly, IArrayOf<IConstTpDropZone>& list)
 {
+
+#ifdef _CONTAINERIZED
+    IPropertyTree * storage = queryGlobalConfig().queryPropTree("storage");
+    if (storage)
+    {
+        StringBuffer pname("planes[@isDropZone='true']");
+        if (name && *name)
+            pname.appendf("[@name='%s']", name);
+        Owned<IPropertyTreeIterator> planes = storage->getElements(pname.str());
+        ForEach(*planes)
+        {
+            IPropertyTree & plane = planes->query();
+            const char * dropzonename = plane.queryProp("@name");
+            if (isEmptyString(dropzonename))
+                continue;
+            const char * prefix = plane.queryProp("@prefix");
+            StringBuffer path(prefix);
+            Owned<IEspTpDropZone> dropZone = createTpDropZone();
+            dropZone->setName(dropzonename);
+            dropZone->setDescription(""); //TODO
+            dropZone->setPath(path);
+            dropZone->setBuild("");
+            dropZone->setECLWatchVisible(true);
+            StringBuffer networkAddress;
+            IArrayOf<IEspTpMachine> tpMachines;
+            Owned<IEspTpMachine> machine = createTpMachine();
+            IpAddress ipAddr;
+            ipAddr.ipset("localhost");
+            ipAddr.getIpText(networkAddress);
+            machine->setNetaddress(networkAddress.str());
+            machine->setConfigNetaddress("localhost");
+            machine->setDirectory(path);
+            machine->setOS(getPathSepChar(path) == '/' ? MachineOsLinux : MachineOsW2K);
+            tpMachines.append(*machine.getLink());
+            dropZone->setTpMachines(tpMachines);
```

Comment:
what would happen if this was left unset?
"localhost"/IP isn't right if dropzone plane is a URL, there is no machine in that context.

Reply:
If the machine is left out, EclWatch despray doesn't display the dropzone.